### ISSUE TITLE
TCP Server example throws exception 

### ISSUE THREAD

**Issue by supercamel** [Cylix/tacopie#16]
Hello, nice library. I'd like to use it but straight away I have encountered this problem. 
Built OK under MSYS2 on Windows 7 with Mingw64 compiler. 

_terminate called after throwing an instance of 'tacopie::tacopie_error'
  what():  fail socket()_

That's the TCP server example copied and pasted. The issue is in the tcp_server constructor.

**Comment by Cylix** [comment]
Hi,

Thanks for reporting this issue!
I just doubled check the tcp_server example, and I noticed I have not included the call to WSAStartup which is necessary on Windows.

Before investigating if there is an issue in tacopie itself, I just want to make sure you did call WSAStartup in your example?

Basically, you would need to do something like this before using anything related to tacopie:

```cpp
  //! Windows netword DLL init
  WORD version = MAKEWORD(2, 2);
  WSADATA data;

  if (WSAStartup(version, &data) != 0) {
    std::cerr << "WSAStartup() failure" << std::endl;
    return -1;
  }
```

and you would also need to call the following at the end of the execution of your program:


```cpp
  WSACleanup();
```

These are windows requirement to initialize networking and tacopie does not call them internally. You would need to include `<Winsock2.h>` to use them.

I'll update the examples to include these calls as it might be misleading otherwise.
Just let me know if it solves your issue :)

Best!
